### PR TITLE
Add test cases for covering actionlib feature

### PR DESCRIPTION
**Pull request by qiuzhong**

### CODE REVIEW

**Comment by qiuzhong** [comment]
Please don't merge this before #370 is merged.

**Comment by minggangw** [comment]
Hi @qiuzhong as the `actionlib` has been merged into `develop`, would you please rebase this PR on `develop` branch and update it, thanks!

**Comment by qiuzhong** [comment]
I'll update it soon.

File: scripts/generate_messages.js
Comment (minggangw) [inline scripts/generate_messages.js]: Please do not change this file, as `npm install` will execute it during installation. We'd better not change the `process.env.AMENT_PREFIX_PATH` variable, the paths added here are only for testing.

I guess you also have to make change to the `run_test.js`

File: scripts/compile_tests.js
Comment (minggangw) [inline scripts/compile_tests.js]: Would you please abstract and wrap a function which can build and install a existing packages. So the `rclnodejs_test_msgs` can share the function with actionlib.

File: test/test-actionlib.js
Comment (minggangw) [inline test/test-actionlib.js:75]: I saw the time out happened on CI when running this case, Could you try to execute `sendGoal` periodically OR set an interval to avoid the contention.

File: scripts/generate_messages.js
Comment (qiuzhong) [reply scripts/generate_messages.js]: @minggangw , the main consideration of changing the `AMENT_PREFIX_PATH` is to generate js files for package `rclnodejs_test_msgs` and `ros1_actions` before testing. If this change must be removed from `npm install` then I have to generate the js packages twice. To make the test work, it's better to move the change to `compile_app.js`.

File: scripts/compile_tests.js
Comment (qiuzhong) [reply scripts/compile_tests.js]: I'll do that.

File: test/test-actionlib.js
Comment (qiuzhong) [reply test/test-actionlib.js:75]: I'll take a try.

File: scripts/generate_messages.js
Comment (minggangw) [reply scripts/generate_messages.js]: For the current implement, all the JavaScript messages generated from IDL files are going to be removed before the running of test, please check [here](https://github.com/RobotWebTools/rclnodejs/blob/develop/scripts/run_test.js#L25)

``` JavaScript
fs.remove(path.join(path.dirname(__dirname), 'generated'), (err)=> {
  if (!err) {
    let mocha = new Mocha();
const testDir = path.join(__dirname, '../test/');
```

So I think you could append the path to `AMENT_PREFIX_PATH` in line [23](https://github.com/RobotWebTools/rclnodejs/blob/develop/scripts/run_test.js#L23). The messages will be re-generated before the test begins.

Comment (qiuzhong) [reply scripts/generate_messages.js]: I noticed that. However, the `ros1_action` package is a little tricky: it needs two rounds. 1st round will generate JavaScript message files and `CMakeLists.txt`. 2nd round will build the native libraries based on the `CMakeLists.txt`.  Maybe I should submit the generated `CMakeLists.txt` just like `rclnodejs_test_msgs`?

Comment (minggangw) [reply scripts/generate_messages.js]: I see and I suggest you could have the following steps:

* rclnodejs.regenerateAll() to generate the JS messages compulsively (a bunch of `.msg` and `CMakeLists.txt` will be present).
* Compile the `ros1_action` project by `CMakeList.txt` generated in step 1 and copy the installation files to the path of ROS 2.
*  Run mocha

Would you please investigate it to check if it's feasible? Thanks!

Comment (qiuzhong) [reply scripts/generate_messages.js]: Sure! I think it should work and I'll try it.

**Comment by minggangw** [comment]
I checked the error from circleci & travis-ci, it seems that the rclnodejs can not find the libraries compiled from the package (action). Please double check whether the libraries are situated at the wanted folder, thanks!

**Comment by qiuzhong** [comment]
The problem is the copying actions of `ros1_actions` and `rclnodejs_test_msgs` to ros2 failed. I'm investigating this.

**Comment by qiuzhong** [comment]
@minggangw , CI errors fixed.

**Comment by minggangw** [comment]
Great, I will merge it when the CI checking passes. Thanks!

**Review by minggangw — APPROVED** [review]
LGTM, good work 👍

**Pull request merged**
